packages/peregrine/src/hooks/useRestApi.js

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useMemo } from 'react';
+
+import * as RestApi from '../RestApi';
+import { useRestResponse } from './useRestResponse';
+
+const { request } = RestApi.Magento2;
+
+export const useRestApi = endpoint => {
+    const [restResponseState, restResponseApi] = useRestResponse();
+    const { receiveError, receiveResponse } = restResponseApi;
+
+    // Define a callback that performs a query
+    // either as an effect or in response to user interaction.
+    const sendRequest = useCallback(async ({ options }) => {
+        try {
+            const response = await request(endpoint, options);
```

Comment:
This hook could call `setLoading` before and after the request so that calling components don't have to worry about it.